chainer/training/non_trainer_extensions/progress_utility.py

```diff
@@ -0,0 +1,199 @@
+from __future__ import print_function
+from __future__ import division
+import time
+
+from chainer.training.extensions import ProgressBarPrinter
+from chainer import utils
+
+
+class TimerUtility(object):
+
+    """Timer utility
+
+    This is a callable utility that will always return True if called at a
+    rate slower than the specified interval. However, if it is called at a
+    faster rate, it will only return True at a maximum rate of once every
+    `interval_seconds` seconds.
+
+    This functionality can be used, for example, when printing information
+    to the console inside a training loop so as to limit the rate at which
+    information is displayed to the user.
+
+    Args:
+        interval_seconds (float): The time interval corresponding to the
+            maximum rate at which this object can return True when called.
+
+    Attributes:
+        iteration: The number of times this object has been called. The
+            initial value is 0.
+        speed: The average number of times per second this object has been
+            called since it was created.
+    """
+
+    def __init__(self, interval_seconds=2):
+        utils.experimental('chainer.training.non_trainer_extensions.TimerUtility')  # NOQA
+        self.iteration = 0
+        self.speed = 0
+
+        self._start_time = time.time()
+        self._interval_seconds = interval_seconds
+        self._last_time = time.time()
+
+    def __call__(self):
+        """Return True or False depending on elapsed time since last call.
+
+        This method should be called once each iteration so that the
+        `iteration` and `speed` attributes will be updated appropriately.
+
+        If at least `interval_seconds` has elapsed since the last call, return
+        True. Otherwise return False.
+        """
+        self.iteration += 1
+        ret_val = False
+        last_time = self._last_time
+        now = time.time()
+        delta_time = now - last_time
+        self.speed = float(self.iteration) / (now - self._start_time)
+        if delta_time >= self._interval_seconds:
+            self._last_time = now
+            ret_val = True
+
+        return ret_val
+
+
+class ProgressUtility(TimerUtility):
```

Comment:
Do we really need to use inheritance here? Using a member variable is enough simple i feel. How do you think?